fog/ocall_oram_storage/trusted/src/lib.rs

```diff
@@ -571,3 +597,775 @@ extern "C" {
+    pub fn release_ocall(_id: u64) {}
+
+    // Mocks the checkout ocall
+    pub fn checkout_ocall<
+        DataSize: ArrayLength<u8> + PartialDiv<U8>,
+        MetaSize: ArrayLength<u8> + PartialDiv<U8>,
+    >(
+        id: u64,
+        idx: &[u64],
+        data: &mut [A64Bytes<DataSize>],
+        meta: &mut [A8Bytes<MetaSize>],
+    ) {
+        debug_assert!(idx.len() == data.len());
+        debug_assert!(idx.len() == meta.len());
+
+        let allocations = ALLOCATIONS.lock().unwrap();
+        let allocation = &allocations[id as usize - 1];
+
+        debug_assert!(allocation.data_size == DataSize::U64);
+        debug_assert!(allocation.meta_size == MetaSize::U64);
+
+        for (x, idx) in idx.iter().enumerate() {
+            let byte_index = (idx * DataSize::U64) as usize;
+            (&mut data[x])
+                .copy_from_slice(&allocation.data[byte_index..byte_index + DataSize::USIZE]);
+            let byte_index = (idx * MetaSize::U64) as usize;
+            (&mut meta[x])
+                .copy_from_slice(&allocation.meta[byte_index..byte_index + MetaSize::USIZE]);
+        }
+    }
+
+    // Mocks the checkin ocall
+    pub fn checkin_ocall<
+        DataSize: ArrayLength<u8> + PartialDiv<U8>,
+        MetaSize: ArrayLength<u8> + PartialDiv<U8>,
+    >(
+        id: u64,
+        idx: &[u64],
+        data: &[A64Bytes<DataSize>],
+        meta: &[A8Bytes<MetaSize>],
+    ) {
+        debug_assert!(idx.len() == data.len());
+        debug_assert!(idx.len() == meta.len());
+
+        let mut allocations = ALLOCATIONS.lock().unwrap();
+        let allocation = &mut allocations[id as usize - 1];
+
+        debug_assert!(allocation.data_size == DataSize::U64);
+        debug_assert!(allocation.meta_size == MetaSize::U64);
+
+        for (x, idx) in idx.iter().enumerate() {
+            let byte_index = (idx * DataSize::U64) as usize;
+            allocation.data[byte_index..byte_index + DataSize::USIZE]
+                .copy_from_slice(data[x].as_ref());
+            let byte_index = (idx * MetaSize::U64) as usize;
+            allocation.meta[byte_index..byte_index + MetaSize::USIZE]
+                .copy_from_slice(meta[x].as_ref());
+        }
+    }
+
+    // Test what happens when we exercise the ORAM
+    // This is simlar to the integration test in `mc-fog-ocall-oram-storage-testing`
+    #[test]
+    fn exercise_oram_storage_shims() {
+        let mut rng = get_seeded_rng();
+        type StorageType = OcallORAMStorage<U1024, U16>;
+
+        // Set tree-top level to 1KB
+        TREETOP_CACHING_THRESHOLD_LOG2.store(10, Ordering::SeqCst);
+
+        let mut st = StorageType::new(131072, &mut rng);
+
+        let mut data_scratch = vec![A64Bytes::<U1024>::default(); 17];
+        let mut meta_scratch = vec![A8Bytes::<U16>::default(); 17];
+
+        // Write 1's along branch at 131072 - 1
+        {
+            st.checkout(131072 - 1, &mut data_scratch, &mut meta_scratch);
```

Comment:
This `131072` repeats a lot; let's make it a constant and document its significance -- why 128 KiB?

Reply:
i think i just needed a number